features/features_test.go

```diff
@@ -0,0 +1,94 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package features_test
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/features"
+)
+
+func Test(t *testing.T) { TestingT(t) }
+
+type featureSuite struct{}
+
+var _ = Suite(&featureSuite{})
+
+func (*featureSuite) TestName(c *C) {
+	c.Check(features.Layouts.String(), Equals, "layouts")
+	c.Check(features.ParallelInstances.String(), Equals, "parallel-instances")
+	c.Check(features.Hotplug.String(), Equals, "hotplug")
+	c.Check(features.SnapdSnap.String(), Equals, "snapd-snap")
+	c.Check(features.PerUserMountNamespace.String(), Equals, "per-user-mount-namespace")
+	c.Check(func() { _ = features.SnapdFeature(1000).String() }, PanicMatches, "unknown feature flag code 1000")
+}
+
+func (*featureSuite) TestKnownFeatures(c *C) {
+	// Check that known features have names.
+	for _, f := range features.KnownFeatures() {
+		c.Check(f.String(), Not(Equals), "", Commentf("feature code: %d", int(f)))
```

Comment:
this should probably check the length as well as I mentioned before

Reply:
Done